quantumdb-postgresql/src/main/java/io/quantumdb/core/planner/TableDataMigrator.java

```diff
@@ -229,6 +233,11 @@ else if (c == '"' && prev != '\\') {
 
 		Map<String, Object> identity = Maps.newHashMap();
 		List<Column> primaryKeyColumns = from.getPrimaryKeyColumns();
+
+		if (parts.size() != primaryKeyColumns.size()) {
```

Comment:
If a table only has 1 primary key, it will return () when no record is returned, this will result in parts being empty while primaryKeyColumns.size() equals 1.
Additionally if (,,,) is returned for 4 primary key columns it will only parse as 3 empty values, therefore 3 != 4 and will return nothing.

Reply:
Can you clarify this one? 

- A table can at most only have one primary key. Did you mean a PK with just one column?
- When this method is called, there's always a record. Did you mean that perhaps `result` might be an empty string? If that's the case this change makes sense.

Reply:
Ah yes, I keep saying 1, 2, 3 or 4 primary keys. What I mean is 1 primary key with that many columns.

Result might be "empty" (consisting of a string with only comma's) if the migration function tries to migrate data that was recently deleted by clients using the database. HammerDB inserts/updates/deletes a lot of entries, that is when I noticed this bug.

The result can be the following strings: 
"()" for a primary key with 1 column.
"(,)" for a primary key with 2 columns.
"(,,)" for a primary key with 3 columns.
...

The function will remove the brackets and then split everything on the comma's that are not escaped. But the function also includes the check to see if the last value is more than length 0. If it is not it will exclude it from the list (I think this might be a bug on its own). All columns in a primary key can never be NULL so I thought that it would be fine to check if the sizes of both lists (primary key columns and the returned values) would be equal, but I think a varchar primary key column can include the empty string "" as a value making that a possible bug.

Reply:
I'm trying to think of what situation could cause one of these cases. I suppose an empty string could result in this, but I doubt HammerDB creates records with such PK values (so that might not be the reason). The change seems logical, I'm just a bit worried we're missing something / not fully understand it.